packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.js

```diff
@@ -25,28 +28,68 @@ const defaultProps = {
   isHovered: false,
   component: 'a',
   isDisabled: false,
-  href: '#'
+  href: '#',
+  keyHandler: () => undefined,
+  index: -1,
+  sendRef: () => undefined
 };
 
-const DropdownItem = ({ className, children, isHovered, component: Component, isDisabled, ...props }) => {
-  const additionalProps = props;
-  if (Component === 'a') {
-    additionalProps['aria-disabled'] = isDisabled;
-    additionalProps.tabIndex = isDisabled ? -1 : additionalProps.tabIndex;
-  } else if (Component === 'button') {
-    additionalProps.disabled = isDisabled;
+class DropdownItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.ref = React.createRef();
+    this.onKeyDown = this.onKeyDown.bind(this);
   }
-  return (
-    <li>
-      <Component
-        {...additionalProps}
-        className={css(isDisabled && styles.modifiers.disabled, isHovered && styles.modifiers.hover, className)}
-      >
-        {children}
-      </Component>
-    </li>
-  );
-};
+
+  componentDidMount() {
+    this.props.sendRef(this.props.index, this.ref, this.props.isDisabled);
+  }
+
+  onKeyDown = event => {
+    // Detected key press on this item, notify the menu parent so that the appropriate
+    // item can be focused
+    if (event.key === 'Tab') return;
+    event.preventDefault();
+    if (event.key === 'ArrowUp') {
+      this.props.keyHandler(this.props.index, 'up');
+    } else if (event.key === 'ArrowDown') {
+      this.props.keyHandler(this.props.index, 'down');
+    }
+  };
+
+  render() {
+    const {
+      className,
+      children,
+      isHovered,
+      keyHandler,
+      component: Component,
+      isDisabled,
+      index,
+      sendRef,
+      ...props
+    } = this.props;
+    const additionalProps = props;
+    if (Component === 'a') {
+      additionalProps['aria-disabled'] = isDisabled;
+      additionalProps.tabIndex = isDisabled ? -1 : additionalProps.tabIndex;
+    } else if (Component === 'button') {
+      additionalProps.disabled = isDisabled;
+    }
+    return (
+      <li>
+        <Component
+          {...additionalProps}
+          className={css(isDisabled && styles.modifiers.disabled, isHovered && styles.modifiers.hover, className)}
+          onKeyDown={this.onKeyDown}
+          ref={ref => (this.ref = ref)}
```

Comment:
Since you are using createRef you can simply pass this.ref to it. Then you can access the current ref with this.ref.current